src/hooks/useContract.ts

```diff
@@ -145,11 +147,21 @@ export function useMainnetInterfaceMulticall() {
 }
 
 export function useV3NFTPositionManagerContract(withSignerIfPossible?: boolean): NonfungiblePositionManager | null {
-  return useContract<NonfungiblePositionManager>(
+  const { account } = useWeb3React()
+  const contract = useContract<NonfungiblePositionManager>(
     NONFUNGIBLE_POSITION_MANAGER_ADDRESSES,
     NFTPositionManagerABI,
     withSignerIfPossible
   )
+  useEffect(() => {
+    if (contract && withSignerIfPossible !== false && account) {
+      sendAnalyticsEvent(InterfaceEventName.WALLET_PROVIDER_USED, {
+        source: 'useV3NFTPositionManagerContract',
+        contract,
+      })
+    }
+  }, [account, contract, withSignerIfPossible])
```

Comment:
```suggestion
  useEffect(() => {
    if (contract && account) {
      sendAnalyticsEvent(InterfaceEventName.WALLET_PROVIDER_USED, {
        source: 'useV3NFTPositionManagerContract',
        contract,
      })
    }
  }, [account, contract])
```

Reply:
when `withSignerIfPossible` is undefined in `useContract` it will be defaulted to `true` so we want to capture that case, that's why the check is explicitly only ignoring if `withSignerIfPossible === false`

Reply:
why does that case matter here for the analytics?